### PR TITLE
upcoming: [M3-7842] - Update Assign Linode Drawer and improve query skipping

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
Pretty simple PR to update the assign linode drawer. Due to some recent UX changes, the decision was made to assign one linode at a time in the drawer and remove the current experience with the unassign selectable list inside the drawer.

## Changes  🔄
- Simplify Assign Linode Drawer logic and adjust tests
- Rename "edit" to "assign" through the feature
- Improve query skipping where relevant

## Preview 📷
localhost_3000_placement-groups_1_linodes_assign

### Prerequisites
- Turn on both the "Placement Groups" feature flag and MSW

### Verification steps
- Navigate to http://localhost:3000/placement-groups/1/linodes/assign
- Click on "Assign Linode to Placement Group"
- Confirm Drawer appearance and behavior

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.test.tsx [hunk 1]
```diff
@@ -57,19 +57,15 @@ describe('PlacementGroupsAssignLinodesDrawer', () => {
...
         selectedPlacementGroup={placementGroupFactory.build()}
       />
     );
 
-    expect(
-      getByText(
-        'There was a problem retrieving your placement group. Please try again'
-      )
-    ).toBeInTheDocument();
+    expect(container).toBeEmptyDOMElement;
```
Comment (bnussman-akamai) [inline packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.test.tsx:68]:
```suggestion
    expect(container).toBeEmptyDOMElement();
```

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx
Comment (bnussman-akamai) [inline packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx:1]: The first time this drawer opens, the open animation does not happen.

https://github.com/linode/manager/assets/115251059/167342e3-0156-45a8-b26b-490ba3bc4e75

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx [hunk 2]
```diff
@@ -138,107 +115,79 @@ export const PlacementGroupsAssignLinodesDrawer = (
...
-          : 'An error occurred while removing the Linode from the group'
+          : 'An error occurred while assigning the Linode to the group'
       );
     }
   };
 
   return (
-    <Drawer onClose={onClose} open={open} title={drawerTitle}>
+    <Drawer onClose={handleDrawerClose} open={open} title={drawerTitle}>
       <Grid>
```
Comment (bnussman-akamai) [inline packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx:138]: Can we remove this `<Grid />`?

File: packages/manager/src/hooks/usePlacementGroupsData.ts [hunk 3]
```diff
@@ -31,9 +31,7 @@ export const usePlacementGroupData = ({
   const { data: linodes, error, isLoading } = useAllLinodesQuery(
     {},
     {
-      '+or': placementGroup?.linodes.map(({ linode }) => ({
-        linode,
-      })),
+      '+or': placementGroup?.linodes.filter(({ linode }) => linode),
```
Comment (bnussman-akamai) [inline packages/manager/src/hooks/usePlacementGroupsData.ts:34]: I think this is the filter we want here

```suggestion
      '+or': placementGroup?.linodes.map((linodeItem) => ({
        id: linodeItem.linode,
      })),
```

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx [hunk 4]
```diff
@@ -41,94 +42,70 @@ export const PlacementGroupsAssignLinodesDrawer = (
           region: selectedPlacementGroup?.region,
```
Comment (bnussman-akamai) [inline packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx:42]: Can we remove the `+or` from this filter. We don't need an `or` if there is only one condition.

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx [hunk 5]
```diff
@@ -41,94 +42,70 @@ export const PlacementGroupsAssignLinodesDrawer = (
...
     allPlacementGroups
   );
 
+  const getLinodeSelectOptions = (): Linode[] => {
+    // We filter out Linodes that are already assigned to a Placement Group
+    return (
+      allLinodesInRegion?.filter((linode) => {
+        return !linodesFromAllPlacementGroups.includes(linode.id);
+      }) ?? []
+    );
```
Comment (bnussman-akamai) [inline packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx:88]: Optional: We could use an API filter to filter out the Linodes already in the placement group. Might be an over-optimization, but we could if you want.

**Review by bnussman-akamai — CHANGES_REQUESTED** [review]

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.test.tsx (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.test.tsx:68]: 🤦

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx
Comment (abailly-akamai) [reply packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx:1]: fixed by not skipping the query in the drawer. this drawer/modal/query business is unsavory at best

File: packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx (hunk 5 above)
Comment (abailly-akamai) [reply packages/manager/src/features/PlacementGroups/PlacementGroupsAssignLinodesDrawer.tsx:88]: I'll make a not of that one as a follow up to do when we are against a real API

File: packages/manager/src/hooks/usePlacementGroupsData.ts (hunk 3 above)
Comment (abailly-akamai) [reply packages/manager/src/hooks/usePlacementGroupsData.ts:34]: yup that seems about right - not for later - optimize queries when working with real data

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/src/features/PlacementGroups/PlacementGroupsUnassignModal.tsx [hunk 6]
```diff
@@ -26,7 +26,10 @@ export const PlacementGroupsUnassignModal = (props: Props) => {
     isLoading,
     mutateAsync: unassignLinodes,
   } = useUnassignLinodesFromPlacementGroup(+placementGroupId ?? -1);
-  const { data: selectedLinode } = useLinodeQuery(+linodeId ?? -1);
+  const { data: selectedLinode } = useLinodeQuery(
+    +linodeId ?? -1,
```
Comment (jaalah-akamai) [inline packages/manager/src/features/PlacementGroups/PlacementGroupsUnassignModal.tsx:30]: Nit: As we come across these in the code, we should just make this more readable using `Number()` like we've been doing for booleans. I dislike this as much as the `!!` operator. (my two cents lol)

**Review by jaalah-akamai — APPROVED** [review]
Looks good and clean

**Pull request merged**